### PR TITLE
Waiting period

### PR DESCRIPTION
**Pull request by treykris**
Links
-----
- Server repo PR: https://github.com/UPchieve/server/pull/267

Description
-----------
- Add a 5 minute waiting period for when students cancel their requests

As the PR stands right now - whenever a session ends (even when matched with a volunteer) a student as to wait 5 minutes from their latest session before making another requesting session.  
The UI currently only disables the button until the appropriate amount of time has past since their last session

Developer self-review checklist
-------------------------------
- [x] Potentially confusing code has been explained with comments
- [x] No warnings or errors have been introduced; all known error cases have been handled
- [x] Any appropriate documentation (within the code, README.md, docs, etc) has been updated
- [x] All edge cases have been addressed

### CODE REVIEW

File: src/views/DashboardView/StudentDashboard/SubjectSelection/SubjectCard.vue [hunk 1]
```diff
@@ -67,9 +67,17 @@ export default {
...
   data() {
     return {
-      selectedSubtopic: ""
+      selectedSubtopic: "",
+      timeoutId: null,
+      disableButton: false
     };
   },
+  mounted() {
+    this.checkOrEnforceWaitingPeriod();
```
Comment (treykris) [inline src/views/DashboardView/StudentDashboard/SubjectSelection/SubjectCard.vue]: @jkeat ,

I spent way more time than I should have deciding on a name for this function, lol

File: src/services/SessionService.js [hunk 2]
```diff
@@ -101,5 +101,26 @@ export default {
...
+
+  getLatestSession(context, user) {
+    return NetworkService.latestSession(context, { user_id: user._id })
+      .then(resp => {
+        const { sessionId, data } = resp.data || {};
+        const { type, subTopic } = data;
+
+        if (type && subTopic && sessionId) {
+          this.currentSession.sessionId = sessionId;
+          this.currentSession.data = data;
```
Comment (jkeat) [inline src/services/SessionService.js]: Is there a need to store this data on the service's `currentSession` prop? It'd be nice to only store it in Vuex

Comment (jkeat) [reply src/services/SessionService.js]: Just to add to this change request -- could you add a visual indicator to tell the student why the button is disabled? Don't need a countdown, just a static message

Comment (treykris) [reply src/services/SessionService.js]: @jkeat,

I went with adding a header (similar to the one when there's a current session active) to display a static message. It's just letting the user know how much time they have left before they can request a new session. The amount of minutes gets updated when the StudentCard component is mounted

Screen Shot 2020-03-31 at 12 24 22 PM

File: src/views/SessionView/index.vue [hunk 3]
```diff
@@ -77,6 +77,9 @@ export default {
 
     window.addEventListener("resize", this.handleResize);
   },
+  beforeDestroy() {
+    window.removeEventListener("resize", this.handleResize);
```
Comment (treykris) [inline src/views/SessionView/index.vue]: @jkeat,

Removing the event listener here helps get rid of the "Back to dashboard" header when we're out of the SessionView

ss

**Pull request merged**